orttraining/orttraining/training_ops/cpu/tensor/gather_nd_grad.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/training_ops/cpu/tensor/gather_nd_grad.h"
+#include "core/providers/cpu/tensor/gather_nd.h"
+#include "core/common/common.h"
+namespace onnxruntime {
+
+#ifndef DISABLE_CONTRIB_OPS
+
+namespace contrib {
+ONNX_OPERATOR_KERNEL_EX(GatherNDGrad, kMSDomain, 1, kCpuExecutionProvider,
+                        KernelDefBuilder()
+                            .TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
+                                                  DataTypeImpl::GetTensorType<double>()})
+                            .TypeConstraint("Tind", {DataTypeImpl::GetTensorType<int64_t>(),
+                                                     DataTypeImpl::GetTensorType<int32_t>()}),
+                        GatherNDGrad);
+
+}  // namespace contrib
+
+#endif
+
+template <typename InputT>
+struct GatherNDGradComputeImpl {
+  void operator()(GatherNDBase::Prepare& p, const Tensor* update_tensor) const {
```

Comment:
on windows build - error C2248: 'onnxruntime::GatherNDBase::Prepare':
cannot access protected struct declared in class 'onnxruntime::GatherNDBase'

Reply:
thanks!, will fix it!!